src/main/kotlin/me/partlysanestudios/partlysaneskies/features/farming/garden/CompostValue.kt

```diff
@@ -0,0 +1,272 @@
+//
+// Written by Su386.
+// See LICENSE for copyright and license notices.
+//
+
+package me.partlysanestudios.partlysaneskies.features.farming.garden
+
+import com.google.gson.JsonParser
+import gg.essential.elementa.UIComponent
+import gg.essential.elementa.components.UIBlock
+import gg.essential.elementa.components.UIWrappedText
+import gg.essential.elementa.constraints.CenterConstraint
+import gg.essential.elementa.dsl.*
+import me.partlysanestudios.partlysaneskies.PartlySaneSkies.Companion.minecraft
+import me.partlysanestudios.partlysaneskies.data.pssdata.PublicDataManager
+import me.partlysanestudios.partlysaneskies.data.skyblockdata.SkyblockDataManager.getItem
+import me.partlysanestudios.partlysaneskies.events.SubscribePSSEvent
+import me.partlysanestudios.partlysaneskies.events.data.LoadPublicDataEvent
+import me.partlysanestudios.partlysaneskies.features.gui.SidePanel
+import me.partlysanestudios.partlysaneskies.render.gui.constraints.ScaledPixelConstraint.Companion.scaledPixels
+import me.partlysanestudios.partlysaneskies.utils.ElementaUtils
+import me.partlysanestudios.partlysaneskies.utils.ElementaUtils.applyBackground
+import me.partlysanestudios.partlysaneskies.utils.MathUtils.round
+import me.partlysanestudios.partlysaneskies.utils.MinecraftUtils.getLore
+import me.partlysanestudios.partlysaneskies.utils.MinecraftUtils.getSeparateUpperLowerInventories
+import me.partlysanestudios.partlysaneskies.utils.StringUtils.formatNumber
+import me.partlysanestudios.partlysaneskies.utils.StringUtils.parseAbbreviatedNumber
+import me.partlysanestudios.partlysaneskies.utils.StringUtils.removeColorCodes
+import me.partlysanestudios.partlysaneskies.utils.StringUtils.stripLeading
+import me.partlysanestudios.partlysaneskies.utils.StringUtils.stripTrailing
+import net.minecraft.client.gui.inventory.GuiChest
+import net.minecraft.inventory.IInventory
+import net.minecraftforge.client.event.GuiScreenEvent
+import java.awt.Color
+import java.util.regex.Matcher
+import java.util.regex.Pattern
+import kotlin.math.ceil
+
+object CompostValue: SidePanel() {
```

Comment:
1. you never check if the option is enabled/disabled
2. the sidebar doesnt appear anymore

Reply:
Fixed both